pkgs/test/tool/host.dart

```diff
@@ -216,36 +201,35 @@ StreamChannel<dynamic> _connectToIframe(String url, int id) {
     if (js_util.getProperty(location, 'href') != iframe.src) return;
 
     message.stopPropagation();
+    windowSubscription.cancel();
 
     if (message.data['ready'] == true) {
       // This message indicates that the iframe is actively listening for
       // events, so the message channel's second port can now be transferred.
-      channel.port2.start();
+      var channel = dom.createMessageChannel();
+      _domSubscriptions[id] = dom.Subscription(channel.port1, 'message',
+          allowInterop((dom.Event event) {
+        controller.local.sink.add((event as dom.MessageEvent).data['data']);
+      }));
+
+  _subscriptions[id] = 
```

Comment:
This formatting is also a bit weird, might want to file an issue if the formatter did this?

Reply:
forgot to reformat.